### ISSUE TITLE
Ability to make changes to an issue as part of an Issue Alert

### ISSUE THREAD

**Issue by snoozbuster** [getsentry/sentry#25741]
## Summary

Right now, Issue Alert Actions allow us to send notifications via email, slack, another integration, or create an issue via Jira. I'd like to be able to make changes to the incoming issue directly as part of the action, including things like:

* Marking an issue as ignored or as "resolved in next release"
* Assigning an issue to a particular person
* Attaching comments to an issue
* Linking a JIRA ticket to an issue (from eg a tag on the issue)

## Motivation

When a frontend engineer opens a PR, our build system automatically deploys that branch to a test URL in our dev environment for review and testing. In these builds, Sentry reporting is enabled (since they are effectively identical to our normal development builds) - so any issues that happen are reported to the oncall via Sentry. This is great for early detection of issues, but tends to be a little bit noisy for the oncall - they have to find the branch owner, send them the issue, make sure it gets resolved, etc etc. I've been able to automate the notification aspect of this (somewhat manually; see also: my comment in [this thread](https://github.com/getsentry/sentry/issues/24795)), but since I can't make any other changes to this issue as part of the alert rule, the issues still show up as part of the "to review" pane in Sentry when really, they don't need any additional review by the oncall - they should be resolved as part of normal feature development at this point. What I would like to set up is:

* WHEN new issue is detected
* IF tag:branch_owner is set
* THEN send message to $branch_owner
* THEN assign issue to $branch_owner
* THEN mark issue as resolved in next release (or even "ignored until next release", which is a separate feature request)
  * alternatively, being able to link a JIRA directly from a tag would solve this as well, as the JIRA for the branch could be used to get the automatic resolution I'm looking for.

The automation resolution is taken in good faith that the engineer will address the issue, and if they don't then it will appear as a regression if the issue is merged into master (which are treated as higher priority by our engineers). Obviously, we can take these steps manually when these issues appear in our "to review" - but that requires some additional training and overhead for each member of the team to do the right steps each time. 

## Additional Context

N/A

**Comment by chadwhitacre** [comment]
Thanks @snoozbuster, routing to @getsentry/workflow for [triage](https://develop.sentry.dev/processing-tickets/#3-triage).

**Comment by scttcper** [comment]
Adding an environment [docs](https://docs.sentry.io/product/sentry-basics/environments/) to the preview environments like staging or pre-prod would allow you to filter the "for review" tab by production issues making it easier for the on-call.

Tags can also be used to assign directly to teams or users via [codeowners](https://docs.sentry.io/product/issues/issue-owners/).  Once the issue has an owner, issue alerts can be setup so that only the issue owner is notified via email, however as you saw in the other thread we don't currently support this via slack.

I'm not sure what is on the roadmap for jira. tagging @getsentry/ecosystem team

**Comment by BYK** [comment]
@snoozbuster does this help you move forward?

**Comment by snoozbuster** [comment]
@BYK @scttcper sorry - a laptop malfunction caused me to be out of commission for the start of the week. Let me look into the codeowners configuration and see what I can do with that - it should help a little but I am still looking for some way to automatically resolve these issues. If I only have issue assigning but not resolution, I am worried that issues which are assigned to a branch owner and then inadvertently make it into master will not get reviewed by the oncall (since they are assigned to another team member).

**Comment by snoozbuster** [comment]
Code owners does work for assigning, but is there any way to avoid one line for each user? this goes on for many lines and doesn't scale - it would be nice if this could just be a single line to the effect of `tags.branch_owner:* ${match}@productiv.com`

![image](https://user-images.githubusercontent.com/5778364/117378939-42d6e980-ae8b-11eb-930d-9cfaf56bce73.png)

**Comment by wedamija** [comment]
I don't think we have any plans to allow dynamic routing based on tag values, at least for the moment. It's a bit of a can of worms - If your DSN is public, then people can tamper with these values as well.
